app/controllers/api/v1/users_controller.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module API
+  module V1
+    class UsersController < ApplicationController
+      include API::V1::ErrorHandlerConcern
+
+      skip_before_action :doorkeeper_authorize!, only: :create
+
+      before_action :ensure_valid_client, only: :create
+
+      def create
```

Comment:
We usually [write](https://github.com/nimblehq/braive-web/blob/development/app/controllers/api/v1/journals_controller.rb#L19) simple `if/else` for rendering success/fail response instead of early return for the controller which make more readable. How about wrap it under `if/else` block?
```Ruby
        user = User.new(create_params.except(:client_id, :client_secret))

        if user.save
          render json: UserTokenSerializer.new(user, { params: { client_id: @client_app.id } }),
                 status: :created
        else
          render json: ActiveModel::ErrorsSerializer.new(user.errors)
        end
```

Reply:
@junan @malparty Yes indeed and it looks cleaner for controller as well, I would prefer going this way as well. Altough @malparty might have been confused with this convention on compass `Prefer using guard clauses over if/else and ternary operators.`, but for controller actions we mostly go for if/else.

Reply:
@malparty I know it might be confusing, but to summarize:

```
    * Prefer using if/else for situation like this (rendering inside controller action)
    * Prefer guard clause for most other scenarios
 ```

Reply:
Also notice that the convention on compass is prefer, it does not say avoid. We can use `if/else` and `guard` clauses based on the readability of code. 
![Uploading Screen Shot 2564-06-23 at 10.10.47.png…]()

Reply:
In fact my confusion came from Rubocop who was not happy with the traditionnal if/else. But this was due to my initial code style -- now with this approach Rubocop allows me a normal if/else. 

It's fixed! ;-) And yeah, clearer ~ 

Thanks!!!